src/DataValues/DecimalValue.php

```diff
@@ -93,33 +85,25 @@ private function assertNumberString( $number ) {
 	 * @throws InvalidArgumentException
 	 */
 	private function convertToDecimal( $number ) {
-		if ( !is_int( $number ) && !is_float( $number ) ) {
-			throw new InvalidArgumentException( '$number must be an int or float.' );
-		}
```

Comment:
Why is nit necessary to remove this constraint? I don't remember why it's here, and it my be good to remove it, but why in this patch?